camel/retrievers/bm25_retriever.py

```diff
@@ -0,0 +1,135 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from typing import Any, Dict, List
+
+import numpy as np
+
+from camel.loaders import UnstructuredIO
+from camel.retrievers import BaseRetriever
+
+DEFAULT_TOP_K_RESULTS = 1
+
+
+class BM25Retriever(BaseRetriever):
+    r"""An implementation of the `BaseRetriever` using the `BM25` model.
+
+        This class facilitates the retriever of relevant information using a
+        query-based approach, it ranks documents based on the occurrence and
+        frequency of the query terms.
+
+        Attributes:
+            bm25 (BM25Okapi): An instance of the BM25Okapi class used for
+                calculating document scores.
+            content_input_path (str): The path to the content that has been
+                processed and stored.
+            chunks (List[Any]): A list of document chunks processed from the
+                input content.
+
+        References:
+            https://github.com/dorianbrown/rank_bm25
+    """
```

Comment:
Maybe you could indent docstring like that? I see all the other functions indented like this.
``` 
    r"""An implementation of the `BaseRetriever` using the `BM25` model.

    This class facilitates the retriever of relevant information using a
    query-based approach, it ranks documents based on the occurrence and
    frequency of the query terms.
  
    Attributes:
        bm25 (BM25Okapi): An instance of the BM25Okapi class used for
            calculating document scores.
        content_input_path (str): The path to the content that has been
            processed and stored.
        chunks (List[Any]): A list of document chunks processed from the
            input content.

    References:
        https://github.com/dorianbrown/rank_bm25
    """
```

Reply:
Yeah, thanks for pointing this out!